apollo-federation/src/sources/connect/validation/extended_type.rs

```diff
@@ -165,6 +165,20 @@ fn validate_field(
         return errors;
     };
 
+    // direct recursion isn't allowed, like a connector on User.friends: [User]
+    if matches!(category, ObjectCategory::Other) && &object.name == field.ty.inner_named_type() {
+        errors.push(Message {
+                    code: Code::CircularReference,
+                    message: format!(
+                        "Direct circular reference detected in `{}.{}: {}`. For more information, see https://go.apollo.dev/connectors/limitations#circular-references",
```

Comment:
This is currently https://preview-docs.apollographql.com/graphos/connectors/limitations/#circular-references-are-unsupported

Reply:
there's a hidden anchor to support a shorter hash: https://github.com/apollographql/private-docs/blob/main/src/content/graphos/connectors/limitations.mdx?plain=1#L89